classes/send_message_form.php

```diff
@@ -0,0 +1,272 @@
+
+        $name = 'subject';
+        $title = get_string('subject', 'tool_messenger');
+        $mform->addElement('textarea', $name, $title, 'wrap="virtual" rows="1" cols="100"');
+        $mform->setType($name, PARAM_TEXT);
+
+        $name = 'message';
+        $title = get_string('message', 'tool_messenger');
+        $mform->addElement('editor', $name, $title,
+            array('enable_filemanagement' => false, 'maxfiles' => 0, 'noclean' => true, 'trusttext' => true));
+        $mform->setType($name, PARAM_RAW);
+
+        $name = 'recipients';
+        $title = get_string('sendto', 'tool_messenger');
+        $roles = $this->get_roles();
+        $mform->addElement('select', $name, $title, $roles)->setMultiple(true);
+        if ($followuprequest) {
+            $mform->disabledIf($name, 'followup', 'neq', 0);
+        }
+        $name = 'knockout_enable';
+        $title = get_string('knockoutenable', 'tool_messenger');
+        $mform->addElement('advcheckbox', $name, $title, ' ');
+        $default = 1;
+        if ($followuprequest) {
+            $default = $parent->get('knockoutdate') == 0;
+            $mform->disabledIf($name, 'followup', 'neq', 0);
+        }
+        $mform->setDefault($name, $default);
+
+        $name = 'knockout_date';
+        $title = get_string ('knockoutdate', 'tool_messenger');
+        $mform->addElement('date_selector', $name, $title, array(
+            'startyear' => 2010,
+            'stopyear'  => 2030,
+            'timezone'  => 99,
+            'optional'  => false
+        ));
+        if ($followuprequest) {
+            $timestamp = (int) $parent->get('knockoutdate');
+            $mform->setDefault($name, $timestamp);
+            $mform->disabledIf($name, 'followup', 'neq', 0);
+        } else {
+            $timestamp = strtotime('-1 years', time());
+            $mform->setDefault($name, $timestamp);
+            $mform->disabledIf($name, 'knockout_enable');
+        }
+
+        $name = 'predictionlink';
+        $title = get_string($name, 'tool_messenger');
+        $linktext = get_string('predicitonlinktext', 'tool_messenger');
+        $mform->addElement('static', '', $title, '<div col-md-9 form-inline felement>'.
+            '<a href="" id="predictiontrigger">'. $linktext . '</a> <div id="predictionbox">' .
+            get_string('noprediction', 'tool_messenger') . '</div>'.
+            '</div>');
+
+        $name = 'priority';
+        $title = get_string('priority', 'tool_messenger');
+        $mform->addElement('select', $name, $title, [1, 2, 3, 4, 5]);
+        $mform->addHelpButton($name, $name, 'tool_messenger');
+
+        $name = 'directsend';
+        $title = get_string('directsend', 'tool_messenger');
+        $mform->addElement('advcheckbox', $name, $title, ' ');
+        $mform->setType($name, PARAM_BOOL);
+
+        // Add Button.
+        $mform->addElement('submit', 'sendmessagebutton', get_string('sendmessagebutton', 'tool_messenger'));
```

Comment:
Maybe some form of return button would be beneficial if the followup view is entered. Otherwise only manually reloading the page will bring you back to the original send message view.